src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java

```diff
@@ -187,16 +188,25 @@ public void purgeQueueAndPersist() {
     }
 
     private void cleanup() {
-        // Delete Event log files belonging to time bucket older than past filesCleanupPeriod(defaults to 60s)
-        long currCleanupTimeBucket = PerformanceAnalyzerMetrics.getTimeInterval(System.currentTimeMillis());
-        if (currCleanupTimeBucket - lastCleanupTimeBucket > filesCleanupPeriodicityMillis) {
-            // Get list of files(time buckets) for purging, considered range : [lastCleanupTimeBucket, currCleanupTimeBucket)
-            List<String> filesForCleanup = LongStream.range(lastCleanupTimeBucket, currCleanupTimeBucket)
-                    .filter(timeMillis -> timeMillis % MetricsConfiguration.SAMPLING_INTERVAL == 0)
-                    .mapToObj(String::valueOf)
-                    .collect(Collectors.toList());
-            eventLogFileHandler.deleteFiles(Collections.unmodifiableList(filesForCleanup));
-            lastCleanupTimeBucket = currCleanupTimeBucket;
+        if (lastCleanupTimeBucket != 0) {
+            // Delete Event log files belonging to time bucket older than past
+            // filesCleanupPeriod(defaults to 60s)
+            long currCleanupTimeBucket =
+                    PerformanceAnalyzerMetrics.getTimeInterval(System.currentTimeMillis());
+            if (currCleanupTimeBucket - lastCleanupTimeBucket > filesCleanupPeriodicityMillis) {
+                // Get list of files(time buckets) for purging, considered range :
+                // [lastCleanupTimeBucket, currCleanupTimeBucket)
+                List<String> filesForCleanup =
+                        LongStream.range(lastCleanupTimeBucket, currCleanupTimeBucket)
+                                .filter(
+                                        timeMillis ->
+                                                timeMillis % MetricsConfiguration.SAMPLING_INTERVAL
+                                                        == 0)
```

Comment:
It seems a little inefficient to test every number. What if we did something like this?
```java
List<String> filesForCleanup = new ArrayList<>();

// ugly expression to handle the case when lastCleanupTimeBucket is divisible by SAMPLING_INTERVAL
// If I did this correctly, then k should satisfy the following: 
//   (k - 1) * SAMPLING_INTERVAL < lastCleanupTimeBucket <= k * SAMPLING_INTERVAL
long k = lastCleanupTimeBucket / SAMPLING_INTERVAL +
        lastCleanupTimebucket % SAMPLING_INTERVAL == 0 ? 0L : 1L;

for (int tick = k * SAMPLING_INTERVAL; tick < currCleanupTimeBucket; tick += SAMPLING_INTERVAL) {
  filesForCleanup.add(String::valueOf(tick));
}

```